Packages/com.llealloo.audiolink/Runtime/Scripts/ThemeColorController.cs

```diff
@@ -16,13 +16,34 @@ public class ThemeColorController : MonoBehaviour
     {
         [UdonSynced] private int _themeColorMode;
 
+        public Color[] customThemeColors
```

Comment:
big unfortunate this is now a footgun property, like
```
controller.customThemeColors[0] = Color.white;
```
will no longer work since you write to a copy. But I guess if it is in the name of optimizing network traffic, it's justified

Reply:
Sadly true, i hadnt thought about this, and that makes it a little weird to work with. Maybe just not using the array at all should be done to prevent confusion

Reply:
Perhaps could mark it as deprecated and change our own code to not use it, if you can be bothered

Reply:
It's fine that you've deprecated this, but please don't make it an error (passing true to ObsoleteAttribute). That would make this a breaking change. Let's save those for later.

Reply:
isnt this somewhat a breaking change though? if people are trying to directly set a index it wont work anymore which is breaking correct?

Reply:
Ah yeah, that's a fair point. It's behaviorally a breaking change. Since we only switched to semver a few releases ago, we haven't talked about if changes like that (where the API doesn't change but the behavior does) should be considered backwards compatible or not (ie bump major or minor version)

Reply:
Regardless it shouldn't need to block this PR

Reply:
wouldnt consider it backwards compatible imo

Reply:
we can mark the setter as deprecated maybe? how good is unities deprecation and updating system

Reply:
From my understanding and research ive done in the past, the popup that can appear when you use built in deprecated code is somehow either hard coded or baked into something in the engine and there doesnt seem to be a attribute to have the same behaviour :(